### DISCUSSION TITLE
Simulator Results Not Being Rendered

### DISCUSSION THREAD

**Discussion by CodyHager** [gorules/jdm-editor#134] [Q&A] [answered]
Hello, I am updating an application to a new version of `jdm-editor` and am having trouble getting the simulator to function as intended. The issue is that when I run a simulation, the output, trace, and input are not shown. I can see that the `onRun` function is being called and everything appears to be working in there. Is there something else I need to do to make sure that the results are rendered?
Using the newest versions of `jdm-editor` and `zen-go`. I have been trying with something like this:
```javascript
  <DecisionGraph
                value={graph}
                onChange={(val) => {
                  setGraph(val)
                }}
                panels={[
                  {
                    id: 'simulator',
                    title: 'Simulator',
                    icon: <PlayCircleOutlined/>,
                    renderPanel: () => (
                      <GraphSimulator
                        onRun={async ({ graph, context }) => {
                          let res = await submitToSim({graph: graph, context})
                          return res;
                        }}
                      />
                    ),
                  },
                ]}
              />
```
<img width="1506" alt="image" src="https://github.com/user-attachments/assets/854aff26-1577-4343-8c95-7fa1acc272e6" />

If someone wouldn't mind telling me how I am setting this up wrong or pointing me towards some documentation, I would appreciate it. Thanks!

**Comment by ivanmiletic** [comment]
Hi, we will take a look

**Comment by CodyHager** [comment, marked as answer]
Figured it out and for anyone else having this problem, you need to pass the graph trace from your engine to the `simulate` field of the `DecisionGraph`. [This](https://github.com/gorules/zen/discussions/194) discussion has code and solved the issue I was having.
